api/session.go

```diff
@@ -179,6 +180,8 @@ func (s *Session) RenewPeriodic(initialTTL string, id string, q *WriteOptions, d
 			// Attempt a session destroy
 			s.Destroy(id, q)
 			return nil
+		case <-ctx.Done():
```

Comment:
If the context is cancelled, seems like the right thing to do here is to also attempt a session.Destroy